### PR TITLE
fix(core): improve redirect behavior after creating account

### PR DESCRIPTION
**Pull request by bc-yevhenii-buliuk**
## What/Why?
This PR improves redirect behavior after creating account, namely, it adds a banner about the successful creation of an account.

## Testing
locally

**before:**

https://github.com/user-attachments/assets/12aaf09a-fc0c-44f3-b69e-d1e93be43131

**after:**

https://github.com/user-attachments/assets/7ff7469d-44b1-431c-978e-f4279fff2221

### CODE REVIEW

**Review by yurytut1993 — APPROVED** [review]
👌

**Review by bc-alexsaiannyi — APPROVED** [review]
looks good

**Comment by bc-yevhenii-buliuk** [comment]
This PR implements appearance of success banner by adding a search parameter to the URL. Thanks to this parameter we can control banner appearance.

Another possible solution to implement was to use the context and change its status to `"success"` after redirecting to the account page, but due to reloading the browser after the redirect, the context returns to its initial state with an `"idle"` status, which does not allow the banner to be displayed.

Open to suggestions, thanks!

**Comment by jorgemoya** [comment]
Hey @bc-yevhenii-buliuk, do you think you can rebase and merge this if test pass? 🙏

**Comment by bc-yevhenii-buliuk** [comment]
> Hey @bc-yevhenii-buliuk, do you think you can rebase and merge this if test pass? 🙏

@jorgemoya yes, sure! I'm making final edits after testing this PR and plan to merge it soon.

File: core/app/[locale]/(default)/account/page.tsx
Comment (bc-yevhenii-buliuk) [reply core/app/[locale]/(default)/account/page.tsx:54]: I think this may be enough, but this message is taken from the design. I will clarify this further with Turner.

File: core/app/[locale]/layout.tsx
Comment (bc-alexsaiannyi) [inline core/app/[locale]/layout.tsx:81]: I wonder if there restrictions to move `AccountStatusProvider` inside `Providers` ?🤔

File: core/app/[locale]/(default)/account/(tabs)/_components/message-wrapper.tsx
Comment (bc-alexsaiannyi) [inline core/app/[locale]/(default)/account/(tabs)/_components/message-wrapper.tsx:11]: What do you think about renaming it? Smth like  `Notification`, `AccountNotification` etc...

File: core/app/[locale]/(default)/account/(tabs)/_components/message-wrapper.tsx
Comment (bc-alexsaiannyi) [inline core/app/[locale]/(default)/account/(tabs)/_components/message-wrapper.tsx:8]: Why do we have this prop as optional?

File: core/app/[locale]/(default)/account/(tabs)/_components/message-wrapper.tsx
Comment (bc-yevhenii-buliuk) [reply core/app/[locale]/(default)/account/(tabs)/_components/message-wrapper.tsx]: I think `AccountNotification` might be a good option, thanks!

**Review by bc-alexsaiannyi — APPROVED** [review]
Looks good. Just don't forget to fix linter errors)

**Review by yurytut1993 — APPROVED** [review]

File: core/app/[locale]/(default)/account/page.tsx
Comment (BC-Turner) [reply core/app/[locale]/(default)/account/page.tsx]: I'm fine with that change. If a partner wants to customize it they can but let's keep it generic.

**Review by bc-azvierieva — APPROVED** [review]
Looks good 👍

**Pull request merged**
